core/services/offchainreporting/OCRContract.go

```diff
@@ -0,0 +1,218 @@
+	if err != nil {
+		return o, errors.Wrap(err, "could not instantiate NewOffchainAggregatorCaller")
+	}
+
+	contractABI, err := abi.JSON(strings.NewReader(offchainaggregator.OffchainAggregatorABI))
+	if err != nil {
+		return o, errors.Wrap(err, "could not get contract ABI JSON")
+	}
+
+	return &OCRContract{
+		ethClient,
+		make(chan offchainaggregator.OffchainAggregatorConfigSet),
+		contractFilterer,
+		contractCaller,
+		address,
+		logBroadcaster,
+		jobID,
+		transmitter,
+		contractABI,
+	}, nil
+}
+
+func (oc *OCRContract) Transmit(ctx context.Context, report []byte, rs, ss [][32]byte, vs [32]byte) error {
+	payload, err := oc.contractABI.Pack("transmit", report, rs, ss, vs)
+	if err != nil {
+		return errors.Wrap(err, "abi.Pack failed")
+	}
+
+	return errors.Wrap(oc.transmitter.CreateEthTransaction(ctx, oc.contractAddress, payload), "failed to send Eth transaction")
+}
+
+func (oc *OCRContract) SubscribeToNewConfigs(context.Context) (ocrtypes.ContractConfigSubscription, error) {
+	sub := &OCRContractConfigSubscription{
+		make(chan ocrtypes.ContractConfig),
+		oc,
+		sync.Mutex{},
+		false,
+	}
+	connected := oc.logBroadcaster.Register(oc.contractAddress, sub)
+	if !connected {
+		return nil, errors.New("Failed to register with logBroadcaster")
+	}
+
+	return sub, nil
+}
+
+func (oc *OCRContract) LatestConfigDetails(ctx context.Context) (changedInBlock uint64, configDigest ocrtypes.ConfigDigest, err error) {
+	opts := bind.CallOpts{Context: ctx, Pending: false}
+	result, err := oc.contractCaller.LatestConfigDetails(&opts)
+	if err != nil {
+		return 0, configDigest, errors.Wrap(err, "error getting LatestConfigDetails")
+	}
+	return uint64(result.BlockNumber), ocrtypes.BytesToConfigDigest(result.ConfigDigest[:]), err
+}
+
+// Conform OCRContract to LogListener interface
+type OCRContractConfigSubscription struct {
+	ch       chan ocrtypes.ContractConfig
+	oc       *OCRContract
+	mutex    sync.Mutex
+	chClosed bool
+}
+
+func (sub *OCRContractConfigSubscription) OnConnect() {}
+func (sub *OCRContractConfigSubscription) OnDisconnect() {
+	sub.mutex.Lock()
+	defer sub.mutex.Unlock()
+
+	if !sub.chClosed {
+		sub.chClosed = true
+		close(sub.ch)
+	}
+}
+func (sub *OCRContractConfigSubscription) HandleLog(lb eth.LogBroadcast, err error) {
+	topics := lb.Log().RawLog().Topics
+	if len(topics) == 0 {
+		return
+	}
+	switch topics[0] {
+	case OCRContractConfigSet:
+		configSet, err := sub.oc.contractFilterer.ParseConfigSet(lb.Log().RawLog())
+		if err != nil {
+			panic(err)
+		}
+		configSet.Raw = lb.Log().RawLog()
+		cc := confighelper.ContractConfigFromConfigSetEvent(*configSet)
+		sub.ch <- cc
+	default:
+	}
+}
+func (sub *OCRContractConfigSubscription) JobID() *models.ID {
+	jobID := sub.oc.jobID
+	return &jobID
+}
+func (sub *OCRContractConfigSubscription) Configs() <-chan ocrtypes.ContractConfig {
+	return sub.ch
+}
+func (sub *OCRContractConfigSubscription) Close() {
```

Comment:
Can this safely be called multiple times?

Reply:
Yes I believe so, although it appears that will also call `OnDisconnect` multiple times.

I defer to @spooktheducks who knows the LogBroadcaster better than I do.

Reply:
I'm of the opinion that you should never be allowed to Open/Start or Close/Stop anything more than once.  I've designed all of the Core code I've written that way.  It panics if you ignore this requirement.

Reply:
I'd argue that Close() is a bit different from Start/Stop/... since the standard library will _usually_ let you close things repeatedly. (I know that strictly speaking the io.Closer interface docs leaves the semantics implementation dependent.)
 Allowing Close() to be called repeatedly without panic (error is fine) also makes life easier for consumers that want to ensure that they're not leaking resources. 
In any case, your call @spooktheducks, but we should clearly document expectations here.

Reply:
I can live with that as long as it's implemented with a `sync.Once` and not something custom

Reply:
@kaleofduty note that in OCR prototype we have several structs in the networking code that WILL panic if you try to close them more than once.